Library/Homebrew/cmd/upgrade.rb

```diff
@@ -104,6 +105,35 @@ def upgrade
         onoe "#{f}: #{e}"
       end
     end
+
+    unless Homebrew.failed?
+      #  Assess existing kegs and reinstall any broken.
+      broken_formulae = []
+
+      CacheStoreDatabase.use(:linkage) do |db|
+        kegs = Formula.installed.collect(&:opt_or_installed_prefix_keg).reject(&:nil?)
+
+        odebug "Assessing keg linkage status" if kegs.size > 1
+
+        kegs.each do |keg|
+          checker = LinkageChecker.new(keg, cache_db: db)
+          next unless checker.broken_library_linkage?
+
+          odebug "#{formula} has broken dynamic library links"
```

Comment:
~~~
Error: undefined local variable or method `formula' for Homebrew:Module
Did you mean?  format
/usr/local/Homebrew/Library/Homebrew/cmd/upgrade.rb:122:in `block (2 levels) in upgrade'
/usr/local/Homebrew/Library/Homebrew/cmd/upgrade.rb:118:in `each'
/usr/local/Homebrew/Library/Homebrew/cmd/upgrade.rb:118:in `block in upgrade'
/usr/local/Homebrew/Library/Homebrew/cache_store.rb:16:in `use'
/usr/local/Homebrew/Library/Homebrew/cmd/upgrade.rb:113:in `upgrade'
/usr/local/Homebrew/Library/Homebrew/brew.rb:87:in `<main>'
~~~

Reply:
the following lets it continue:

~~~
diff --git a/Library/Homebrew/cmd/upgrade.rb b/Library/Homebrew/cmd/upgrade.rb
index 9256e6e..dbc05b2 100644
--- a/Library/Homebrew/cmd/upgrade.rb
+++ b/Library/Homebrew/cmd/upgrade.rb
@@ -119,9 +119,9 @@ module Homebrew
           checker = LinkageChecker.new(keg, cache_db: db)
           next unless checker.broken_library_linkage?
 
-          odebug "#{formula} has broken dynamic library links"
+          odebug "#{keg} has broken dynamic library links"
 
-          broken_formulae << resolve_formula(keg)
+          broken_formulae << Formulary.from_keg(keg)
         end
       end
 
~~~